client/src/main/java/org/asynchttpclient/Request.java

```diff
@@ -180,4 +180,16 @@
    * @return the NameResolver to be used to resolve hostnams's IP
    */
   NameResolver<InetAddress> getNameResolver();
+
+  /**
+   * @return a new request builder using this request as a prototype
+   */
+  RequestBuilder toBuilder();
```

Comment:
This is a breaking change. Anyone with his own implementation like you would be forced to modify his code and recompile. Please make it a default method.

Reply:
Agreed. Moved as default method